### PR TITLE
bugfix: update api server cert sans for all cluster cert

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

1. update api server cert sans for all cluster cert: using `seautil` cmd to do remote cert update action, load local ca cert and apiserver cert, adding alt-names to "Subject Alternative Name" to generate  new one. its mounted to api server, thus, no need to reload apiserver pod.
2. delete update_cert func from cluster runtime interface.
3. make cert fixed name like cert base name or CN name as const value.
4. rename some func name with its real logic.

### CODE REVIEW

File: cmd/seautil/cmd/certs.go [hunk 1]
```diff
@@ -33,25 +33,22 @@ type Flag struct {
...
+func NewCertGenCmd() *cobra.Command {
+	config := new(Flag)
 
-func init() {
-	config = &Flag{}
-	rootCmd.AddCommand(certsCmd)
+	var certsCmd = &cobra.Command{
+		Use:   "gen",
+		Short: "generate kubernetes certs",
+		Long:  `seautil cert gen --node-ip 192.168.0.2 --node-name master1 --dns-domain aliyun.com --alt-names aliyun.local --service-cidr 10.103.97.2/24`,
```
Comment (allencloud) [inline cmd/seautil/cmd/certs.go:43]: No, please remove the aliyun-related details. sealer is a universe tool, it should not add commercial related content in its code.

File: cmd/seautil/cmd/certs.go [hunk 2]
```diff
@@ -33,25 +33,22 @@ type Flag struct {
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			nodeIP := net.ParseIP(config.NodeIP)
+			if nodeIP == nil {
+				return fmt.Errorf("input --node-ip(%s) is an invalid IP format", config.NodeIP)
```
Comment (allencloud) [inline cmd/seautil/cmd/certs.go:47]: s/is/is not/g

File: cmd/seautil/cmd/certs.go [hunk 3]
```diff
@@ -60,4 +57,47 @@ func init() {
...
+	return certsCmd
+}
+
+func NewCertUpdateCmd() *cobra.Command {
+	var altNames []string
+
+	certCmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update Kubernetes API server's cert",
+		Long:  `seautil cert update --alt-names aliyun.local`,
```
Comment (allencloud) [inline cmd/seautil/cmd/certs.go:70]: Same issue, and please remove aliyun-related content.

File: cmd/seautil/cmd/certs.go [hunk 4]
```diff
@@ -60,4 +57,47 @@ func init() {
...
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(altNames) == 0 {
+				return fmt.Errorf("IP address or DNS domain needed for cert Subject Alternative Names")
+			}
+
+			err := cert.UpdateAPIServerCertSans(cert.KubeDefaultCertPath, altNames)
+			if err != nil {
+				return fmt.Errorf("failed to update api server: %+v", err)
```
Comment (allencloud) [inline cmd/seautil/cmd/certs.go:79]: update API server's cert

File: pkg/cert/kube_certs.go [hunk 5]
```diff
@@ -328,3 +313,82 @@ func (meta *MetaData) GenerateAll() error {
 	}
 	return nil
 }
+
+// GenerateCert generate all cert.
+func GenerateCert(certPath, certEtcdPath string, altNames []string, hostIP net.IP, hostName, serviceCIRD, DNSDomain string) error {
+	// if no certPath, certEtcdPath specified, will use default kubernetes pki path "/etc/kubernetes/pki".
+	if certPath == "" {
```
Comment (allencloud) [inline pkg/cert/kube_certs.go:320]: Cert-related work is very very serious in the cluster operation. If a user has no input config, I suggest that we should block the request and must let them input clear and correct cert config.

In addition, I wish that in the command definition layer, we validate if the input cert config is empty.

**Review by allencloud — DISMISSED** [review]
I have left some request change

File: cmd/seautil/cmd/certs.go [hunk 6]
```diff
@@ -33,25 +33,22 @@ type Flag struct {
...
-	config = &Flag{}
-	rootCmd.AddCommand(certsCmd)
+	var certsCmd = &cobra.Command{
+		Use:   "gen",
+		Short: "generate kubernetes certs",
+		Long:  `seautil cert gen --node-ip 192.168.0.2 --node-name master1 --dns-domain sealer.com --alt-names sealer.local --service-cidr 10.103.97.2/24`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			nodeIP := net.ParseIP(config.NodeIP)
+			if nodeIP == nil {
+				return fmt.Errorf("input --node-ip(%s) is not in a valid IP format", config.NodeIP)
```
Comment (allencloud) [inline cmd/seautil/cmd/certs.go:47]: please remove `in`

File: cmd/seautil/cmd/certs.go [hunk 7]
```diff
@@ -60,4 +57,47 @@ func init() {
...
+
+func NewCertUpdateCmd() *cobra.Command {
+	var altNames []string
+
+	certCmd := &cobra.Command{
+		Use:   "update",
+		Short: "Update Kubernetes API server's cert",
+		Long:  `seautil cert update --alt-names sealer.cool`,
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (justadogistaken) [inline cmd/seautil/cmd/certs.go:72]: we can use cmd.MarkFlagRequired("alt-names").

**Review by VinceCui — APPROVED** [review]
LGTM

**Pull request merged**